peer-to-peer-scheduler/peer-to-peer-scheduler.gs

```diff
@@ -0,0 +1,900 @@
+    if (!event) {
+      invalidMeetings.push(rowNumber);
+      continue;
+    }
+    
+    let mentorEmail = activeSheet.getRange(rowNumber, MENTOR_EMAIL_INDEX + 1).getValue().trim();
+    if (!validateEmail(mentorEmail)) {
+      invalidEmails.push({row: rowNumber, email: mentorEmail});
+      continue;
+    }
+
+    let mentorInvited = event.getGuestByEmail(mentorEmail);
+    if (!mentorInvited) {
+      // Do not use addGuest method. It won't send emails to new guests.
+      if (addAttendees(calendarId, meetId, [mentorEmail])) {
+        ++numSuccess;
+      } else {
+        invalidMeetings.push(rowNumber);
+      }
+    } else {
+      ++numSkipped;
+    }
+  }
+
+  let summary = `Number of rows for which invites have been sent: ${numSuccess}
+  Number of rows skipped: ${numSkipped} (invite was already sent previously or intentionally skipped)
+  Number of rows with invalid emails: ${invalidEmails.length}
+  Number of rows with invalid meeting id: ${invalidMeetings.length}
+  Number of rows with no meeting: ${noMeetingRows.length}
+  `;
+  
+  summary += `\n---
+
+  Following rows have invalid email id.
+  `;
+  for (let i = 0; i < invalidEmails.length; ++i) {
+    summary += `\nRow ${invalidEmails[i].row}: ${invalidEmails[i]}.email`;
+  }
+
+  summary += `\n---
+
+  Following rows have invalid meeting id.
+  `;
+  for (let i = 0; i < invalidMeetings.length; ++i) {
+    summary += `\nRow ${invalidMeetings[i]}`;
+  }
+
+  summary += `\n---
+
+  Following rows have no meeting.
+  `;
+  for (let i = 0; i < noMeetingRows.length; ++i) {
+    summary += `\nRow ${noMeetingRows[i]}`;
+  }
+  showInfo(TITLE_SUCCESS, summary);
+}
+
+function getRecordingAndChatLinks() {
+  let setupResult = initSetup();
+  if (!setupResult.success) {
+    return;
+  }
+  let { calendar, activeSheet, oneToOneData } = setupResult;
+
+  let noMeetingRows = [];
+  let invalidMeetings = [];
+  let noRecordingLinks = [];
+  let noChatLinks = [];
+  let numSuccess = 0;
+  let numSkipped = 0;
+  let calendarId = calendar.getId();
+
+  for (let rowIndex = 0; rowIndex < oneToOneData.length; ++rowIndex) {
+    let rowNumber = rowIndex + DATA_BEGIN_ROW;
+    let opsRequestFlag = parseInt(activeSheet.getRange(rowNumber, OPS_REQUEST_INDEX + 1).getValue());
+    if (opsRequestFlag !== FLAG_SEND_INVITE) {
+      // The row should be skipped
+      ++numSkipped;
+      continue;
+    }
+
+    let meetIdCell = activeSheet.getRange(rowNumber, MEETING_ID_INDEX + 1);
+    let meetId = meetIdCell.getValue().trim();
+    if (meetId === "") {
+      // Meeting is not set
+      noMeetingRows.push(rowNumber);
+      continue;
+    }
+
+    let event = getEvent(calendar, meetId);
+    if (!event) {
+      invalidMeetings.push(rowNumber);
+      continue;
+    }
+
+    let instancesIds = [];
+    let instancesArray = getRecurringInstances(calendarId, meetId);
+    if(!instancesArray) {
+      invalidMeetings.push(rowNumber);
+      continue;
+    }
+
+    instancesArray.forEach(
```

Comment:
Please use `for .. of` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
Replaced `forEach` usage with `for...of`